### DISCUSSION TITLE
Saving using Node.js using FileSystemPaths FileSystemExtensions

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#5654] [Q&A] [answered]
I am using in browser plugin development with Tiddlywiki running on Node.JS using the below configs. 

**$:/config/FileSystemPaths**

```
[suffix[.info]!has[draft.of]removesuffix[.info]removeprefix[$:/]addprefix[system/]]
[is[system]!has[draft.of]removeprefix[$:/]addprefix[system/]]
[is[draft]search-replace:g:regexp[/|\\],[_]addprefix[drafts/]]
```

$:/config/FileSystemExtensions  contains

```
[suffix[.info]then[.info]]
```

I have an update button to create or update (overwrite)  the plugin.info for me. The plugin info is created in a tiddler like this

`$:/plugins/kookma/mynewplugin/plugin.info`

Then the file is saved in a proper location using the above configs!

The issue here is I have always one tiddler in my wiki (`$:/plugins/kookma/mynewplugin/plugin.info`) even when I push the update button even several times! but I got these files in my plugin folder

- plugin.info
- plugin.info.meta
- plugin_1.info
- plugin_1.info.meta

I am wondering why I have plugin_1.info when there is no such tiddler in my wiki?

**Comment by joshuafontany** [comment]
Normally, the .info file data is not accessible from within tiddlywiki, so this is interesting. Something is happening there where TW is attempting to write to that location without having loaded that tiddler's data from that location. Did you load the .info file by having an accompanying .info.meta file, or was it loaded in some other way? (A tiddlywiki.files file?)

**Reply by kookma** [reply]
Hi Joshua,
 The file actually is a tiddler with this name: `$:/plugins/kookma/mynewplugin/plugin.info`
  So I expect it to be saved in `my-wiki-tiddlers/system/plugins/kookma/mynewplugin` folder

**Reply by saqimtiaz** [reply]
@kookma the problem is that the plugin.info file already exists and is not a tiddler. Therefore TW cannot and does not overwrite this file and creates a plugin_1.info file instead. TW only overwrites files that were originally present when the wiki started **and loaded as tiddlers**.

**Reply by kookma** [reply]
@saqimtiaz - I use a button to create the plugin.info tiddler first and the use the $:/config/FileSystemExtensions to create the plugin.info file.

To be more clear. I use the ThirdFlow for creating plugins using Node.js but in browser. The ThirdFlow plugin itself does not support creating plugin.info so I had to create it manually!
I wrote a macro on click it creates a `plugin.info` for me on the fly! for example for my new plugin called `mynewplugin`, it creates this tiddler: `$:/plugins/kookma/mynewplugin/plugin.info` it saves it in plugin folder, but then I press the button to update from example revision number and then I noticed there are two plugin.info files in the folder one with _1 prefix.

**Comment by kookma** [comment, marked as answer]
Ah I removed the 

```
[suffix[.info]!has[draft.of]removesuffix[.info]removeprefix[$:/]addprefix[system/]]
```

from $:/config/FileSystemPaths and kept the `$:/config/FileSystemExtensions` intact!

It seems the problem has gone! I think as I used two rules the issue happened!

Thank you Joshua
